doc/source/generation.rst

```diff
@@ -1,5 +1,153 @@
+
+Tree and star
++++++++++++++
+:meth:`Graph.Tree` can be used to generate regular trees, in which almost each vertex has the same number of children:
+
+>>> g = Graph.Tree(n=7, n_children=2)
+
+creates a tree with seven vertices - of which four are leaves. The root (0) has two children (1 and 2), each of which has two children (the four leaves). Regular trees can be directed or undirected (default).
+
+The method :meth:`Graph.Star` creates a star graph, which is a subtype of a tree.
+
+Lattice
++++++++
+:meth:`Graph.Lattice` creates a regular lattice of the chosen size. For instance:
+
+>>> g = Graph.Lattice(dim=[3, 3], circular=False)
+
+creates a 3x3 grid in two dimensions (9 vertices total). `circular` is used to connect each edge of the lattice back onto the other side, a process also known as "periodic boundary condition" that is sometimes helpful to smoothen out edge effects.
+
+The one dimensional, circular case (a ring) is important enough to deserve its own constructor :meth:`Graph.Ring`.
+
+.. note:: The line graph is a one dimensional, non-circular lattice, and can be also constructed as a non-circular ring.
+
+Graph atlas
++++++++++++
+The book ‘An Atlas of Graphs’ by Roland C. Read and Robin J. Wilson contains all undirected graphs with up to seven vertices, numbered from 0 up to 1252. You can create any graph from this list by index with :meth:`Graph.Atlas`, e.g.:
+
+>>> g = Graph.Atlas(44)
+
+The graphs are listed:
+
+ - in increasing order of number of nodes;
+ - for a fixed number of nodes, in increasing order of the number of edges;
+ - for fixed numbers of nodes and edges, in increasing order of the degree sequence, for example 111223 < 112222;
+ - for fixed degree sequence, in increasing number of automorphisms.
+
+
+Famous graphs
++++++++++++++
+A curated list of famous graphs, which are often used in the literature for benchmarking and other purposes, is available on the `igraph C core manual <https://igraph.org/c/doc/igraph-Generators.html#igraph_famous>`_. You can generate any graph in that list by name, e.g.:
+
+>>> g = Graph.Famous('Zachary')
+
+will teach you some about martial arts.
+
+
+Random graphs
++++++++++++++
+Stochastic graphs can be created according to several different models or games:
+
+ - Barabasi-Albert model: :meth:`Graph.Barabasi`
+ - Erdos-Renyi: :meth:`Graph.Erdos_Renyi`
+ - Watts-Strogatz :meth:`Graph.Watts_Strogatz`
+ - stochastic block model :meth:`Graph.SBM`
+ - forest fire game :meth:`Graph.Forest_Fire`
+ - random geometric graph :meth:`Graph.GRG`
+ - growing :meth:`Graph.Growing_Random`
+ - establishment game :meth:`Graph.Establishment`
+ - preference, the non-growing variant of establishment :meth:`Graph.Preference`
+ - asymmetric preference :meth:`Graph.Asymmetric_Prefernce`
+ - recent degree :meth:`Graph.Recent_Degree`
+ - K-regular (each node has degree K) :meth:`Graph.K_Regular`
```

Comment:
Tiny nitpick: mathematicians typically use lower-case $k$ when talking about k-regular graphs. (Of course, some may use any letter if it fits the context.)

Reply:
yep, fixed